### PR TITLE
feat(tests): Ajouter des tests BDD pour le contrôle du préchauffage a…

### LINKED ISSUE #90: [BUG] IHP Enable/Disable Switch ignored - Preheating still starts when Switch is off
## Bug Description

When the _IHP Enable/Disable Switch_ is turned **off**, the Intelligent Heating Pilot still starts preheating cycles.
This occurs reliably across all configured rooms/services, even though preheating should be fully disabled when the switch is off.

## Environment

- **Home Assistant Version**: 2026.1.3
- **IHP Version**: v0.5.0-rc2
- **VTherm Version**: 8.6.2
- **Scheduler component Version**: 3.3.8
- **Scheduler Card Version**: v4.0.11
- **Installation Method**: [x] HACS

## Steps to Reproduce

1. Set up the IHP integration on three VTherm (3 services = 3 rooms = 3 TRVs).
2. Set up Scheduler component.
3. Set up schedule entities with Scheduler Card
   3.1. Room 1 (kitchen):
       - target 18 °C (eco) at 00:00
       - target 21 °C (comfort) at 05:00
       - target 18 °C (eco) at 07:30

   3.2. Room 2 (living room):
       - target 19 °C (eco) at 00:00
       - target 22 °C (comfort) at 07:30
       - target 19°C (eco) at 22:00

   3.3. Room 3 (bathroom):
       - target 18 °C (eco) at 00:00
       - target 21 °C (comfort) at 05:00
       - target 18 °C (eco) at 07:30
       - target 21 °C (comfort) at 16:00
       - target 18 °C (eco) at 18:30

4. Disable Switch to toggle IHP preheating (from on) to off for all three services.
5. IHP preheating...
   5.1. started at 03:40 for Room 1 (current temperature was 19.4 °C and slope 1.21 °C/h).

   5.2. started at 01:47 for Room 2 (current temperature was 20.4 °C and slope 0.28 °C/h).
   
   5.3. started at 03:57 for Room 3 (current temperature was 20.2 °C and slope 0.76 °C/h).
 
## Expected Behavior

If the IHP Enable/Disable Switch is set to off, no preheating should be triggered and all anticipation should be skipped or canceled.

## Actual Behavior

Preheating events were triggered for all three VTherm services despite the Switch showing state **off**.
Preheating start times match IHP prediction logic (e.g., 03:40, 01:47, 03:57), indicating IHP logic is still executing.

## Screenshots

I have added sample screenshots from Room 1 (kitchen).

Image

Image

Image

## Additional Context

It happens consistently for all three services after I turned off the IHP Enable/Disable Switch.

Additional notes to screenshots above for Room 1:

At 03:55:
- I manually set the preset mode back (from 21 °C which is comfort) to 18 °C (eco).

At 05:00:
- The Scheduler component set the preset mode to 21 °C (comfort).

## Possible Solution

<details>
<summary>Click to expand possible solution according to GitHub Copilot (Claude Opus 4.5)</summary>

> _Looking at `event_bridge.py`, the `_recalculate_and_publish` method unconditionally calls `calculate_and_schedule_anticipation()` without any `ihp_enabled` check. However, this shouldn't be the bug if `calculate_and_schedule_anticipation` already handles `ihp_enabled` internally._
>
> _**Other possible causes:**_
> _1. **Scheduler integration itself** - The scheduler entity (e.g., `scheduler-card`) might have its own anticipation/preheating feature enabled independently of IHP._
> _2. `ihp_enabled` **value not updating** - The flag might be cached or not refreshing when you toggle the switch._
> _3. **Race condition** - The `cancel_action` might execute, but the scheduler immediately reschedules before IHP's next check._
> _4. **Multiple IHP instances** - If you have multiple config entries, another instance might be triggering the scheduler._
> _5. `cancel_action` **implementation** - The `cancel_action` in your `SchedulerCommander` might not actually revert the scheduler's time properly._

</details>

### PR DESCRIPTION
**Pull request by RastaChaum**
## Description

Add BDD Tests validating pre-heating switch feature

## Related Issue

Fixes #90

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 📝 Documentation update
- [ ] 🔧 Refactoring (no functional changes)
- [X] ✅ Test coverage improvement
- [ ] 🏗️ Infrastructure/build changes

## Changes Made
- add ihp_enable_disable_switch.feature and test_ihp_enable_disable_switch.py

## Testing Performed

- [X] All existing unit tests pass
- [X] Added new unit tests for changes
- [ ] Tested manually in Home Assistant
- [ ] Tested with multiple VTherm configurations
- [ ] Tested edge cases

### Test Details

**Test Scenarios:**
1. When IHP switch is disabled, no preheating occurs before scheduled time
2. When IHP switch is enabled, intelligent preheating occurs as expected 
3. Disabling IHP switch during active preheating cancels preheating immediately
4. Re-enabling IHP switch after disabling it resumes anticipation calculations
5. Multiple scheduler events respect IHP disable state

## Checklist

- [X] My code follows the project's style guidelines (PEP 8, Black formatting)
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings or errors
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published
- [ ] I have updated CHANGELOG.md in the `[Unreleased]` section
- [X] Domain layer has NO `homeassistant.*` imports (if applicable)
- [X] All external interactions use ABCs/interfaces (if applicable)
- [X] Value objects are immutable with `@dataclass(frozen=True)` (if applicable)

## Architecture Compliance (for code changes)

- [X] Domain layer remains independent of Home Assistant
- [X] Business logic is in domain services, not infrastructure
- [X] All infrastructure changes implement existing interfaces
- [ X] No business rules in adapters
- [ X] Tests use mocks for external dependencies

## Breaking Changes

**Impact:**
- 

**Migration Guide:**

## Reviewer Checklist

- [ ] Code follows project architecture (DDD principles)
- [ ] Tests are comprehensive and pass
- [ ] Documentation is clear and complete
- [ ] Breaking changes are clearly documented
- [ ] CHANGELOG.md is updated
- [ ] Commit messages follow Conventional Commits

### CODE REVIEW

File: tests/features/ihp_enable_disable_switch.feature [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    Then the IHP enable switch should still be in the off state
+    And no preheating should occur after restart
+
+  Scenario: Multiple scheduler events respect IHP disable state
+    Given the IHP enable switch is turned off
+    And scheduler has 3 events today (07:00, 12:00, 19:00)
+    When the system processes all scheduled events
+    Then none of the events should trigger preheating
+    And all events should occur at their exact scheduled times
+    And room temperature should only change at scheduled times
```
[the remark this answers is not in the document]
Comment (RastaChaum) [reply tests/features/ihp_enable_disable_switch.feature]: @copilot are you able to implement these behavior test ?
